pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1016,30 +1016,24 @@ func (pl *dynamicResources) lookupClassParameters(logger klog.Logger, class *res
 }
 
 func (pl *dynamicResources) lookupClaimParameters(logger klog.Logger, class *resourcev1alpha2.ResourceClass, claim *resourcev1alpha2.ResourceClaim) (*resourcev1alpha2.ResourceClaimParameters, *framework.Status) {
-	defaultClaimParameters := resourcev1alpha2.ResourceClaimParameters{
-		Shareable: true,
-		DriverRequests: []resourcev1alpha2.DriverRequests{
-			{
-				DriverName: class.DriverName,
-				Requests: []resourcev1alpha2.ResourceRequest{
-					{
-						ResourceRequestModel: resourcev1alpha2.ResourceRequestModel{
-							// TODO: This only works because NamedResources is
-							// the only model currently implemented. We need to
-							// match the default to how the resources of this
-							// class are being advertized in a ResourceSlice.
-							NamedResources: &resourcev1alpha2.NamedResourcesRequest{
-								Selector: "true",
-							},
-						},
-					},
-				},
-			},
-		},
-	}
+	if claim.Spec.ParametersRef == nil && class.DefaultClaimParametersRef == nil {
+		return nil, statusError(logger, fmt.Errorf("error allocating claim, either claim's parameter reference or default parameters for resourceclass need to be set"))
```

Comment:
and also should we return statusError or statusUnScheduleable?

Reply:
`statusUnscheduleable` - the pod is permanently unschedulable until claim or class are changed or the parameters are created. The plugin reacts to cluster events for that, so there is no need for periodic retries, which is what `statusError` implies.